reproject/interpolation/core_celestial.py

```diff
@@ -23,7 +25,8 @@ def _reproject_celestial(array, wcs_in, wcs_out, shape_out, order=1):
     """
 
     # Make sure image is floating point
-    array = np.asarray(array, dtype=float)
+    if not np.issubdtype(array.dtype, np.float):
+        array = np.asarray(array, dtype=float)
```

Comment:
I think we could also do ``array = array.astype(float)`` which avoids a copy if it isn't needed